engines/sci/engine/script_patches.cpp

```diff
@@ -7140,8 +7140,51 @@ static const uint16 qg4InnPathfindingPatch[] = {
 	PATCH_END
 };
 
+// When autosave is enabled, Glory::save() (script 0) deletes savegame files in
+// a loop: while disk space is insufficient for a new save, or while there are
+// 20+ saves. Since ScummVM handles slots differently and allows far more
+// slots, this deletes all but the most recent 19 manual saves, merely by
+// walking from room to room!
+//
+// Ironically, kGetSaveFiles() (kfile.cpp) and the debugger's 'list_files'
+// command rely on listSavegames() (file.cpp), which specifically omits the
+// autosave slot, so the script will only ever delete manual saves. And the
+// space check doesn't take into account the reduced demand when overwriting an
+// existing autosave.
+//
+// No good can come of this loop. So we skip it entirely. If the disk truly is
+// out of space, a message box will complain, and the player can delete saves
+// voluntarily.
+//
+// Note: Glory::save() contains another space freeing loop, but it might be
+// unreachable.
+//
+// Applies to at least: English CD, English floppy
+// Fixes bug: #10758
+static const uint16 qg4AutosaveSignature[] = {
+	SIG_MAGICDWORD,
+	0x43, 0x3f, SIG_UINT16(0x0002),     // callk CheckFreeSpace[3f], 02
+	0x18,                               // not
+	0x2f, 0x05,                         // bt 05 [skip other OR condition]
+	0x8d, 0x09,                         // lst temp[9] (savegame file count)
+	0x35, 0x14,                         // ldi 20d
+	0x20,                               // ge?
+	0x30, PATCH_UINT16(0x0038),         // bnt [end the loop]
+	SIG_END
+};
+
+static const uint16 qg4AutosavePatch[] = {
+	0x34, PATCH_UINT16(0x0000),         // ldi 0 (waste 12 bytes...)
+	0x34, PATCH_UINT16(0x0000),         // ldi 0
+	0x34, PATCH_UINT16(0x0000),         // ldi 0
+	0x34, PATCH_UINT16(0x0000),         // ldi 0
+	0x32, PATCH_UINT16(0x0038),         // jmp [end the loop]
```

Comment:
Good work,
but please remove the "ldi 0" patches and move the "jmp" upwards plus adjust the offset of the jump.